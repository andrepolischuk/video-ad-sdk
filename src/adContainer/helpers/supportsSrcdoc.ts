const supported = 'srcdoc' in document.createElement('iframe')
const supportsSrcdoc = (): boolean => supported

export default supportsSrcdoc
