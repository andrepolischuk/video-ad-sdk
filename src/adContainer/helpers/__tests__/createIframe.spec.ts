import createIframe from '../createIframe';
import supportsSrcdoc from '../supportsSrcdoc';

jest.mock('../supportsSrcdoc');
describe('createIframe', () => {
  beforeEach(() => {
    // eslint-disable-next-line no-undef
    jsdom.reconfigure({
      url: 'https://www.example.com/'
    });

    supportsSrcdoc.mockReturnValue(false);
  });

  test('must return an empty Iframe element', async () => {
    const iframe = await createIframe(document.body, 'test');

    expect(iframe).toBeInstanceOf(HTMLIFrameElement);
    expect(iframe).toMatchSnapshot();

    iframe.parentElement.removeChild(iframe);
  });

  test('must post a message to notify that is ready', async () => {
    expect.assertions(1);
    const handleMessage = ({data}) => {
      expect(data).toBe('test_iframe_ready');
    };

    window.addEventListener('message', handleMessage, false);
    await createIframe(document.body, 'test_iframe');
  });

  test('must throw (reject) if there is a problem creating the iframe', async () => {
    expect.assertions(1);
    try {
      await createIframe(document.createElement('div'), test);
    } catch (error) {
      expect(error).toBeInstanceOf(Error);
    }
  });
});
