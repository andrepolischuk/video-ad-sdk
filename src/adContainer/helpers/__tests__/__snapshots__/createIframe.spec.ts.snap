// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createIframe must return an empty Iframe element 1`] = `
<iframe
  loading="eager"
  sandbox="allow-forms allow-popups allow-scripts allow-same-origin"
  src="about:blank"
  style="margin: 0px; padding: 0px; width: 0px; height: 0px; position: absolute;"
/>
`;
